### PR TITLE
feat(antd): create themes and layout components to use with themes

### PR DESCRIPTION
**Pull request by alicanerdurmaz**
following changes implemented for `@refinedev/antd`:

-   blue, purple, magenta, red, orange, yellow, and green themes were added.

-   `<ThemedLayout>`, `ThemedSider`, `<ThemedTitle>`, `<ThemedHeader>` created to use theme colors.

-   `<EditButton>` in `<Show>` type changed to `primary`.
-   `CreateButton` type changed to `primary`.

### CODE REVIEW

File: packages/antd/src/components/themedLayout/sider/index.tsx [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+} from "@refinedev/core";
+
+import { drawerButtonStyles } from "./styles";
+import { RefineThemedLayoutSiderProps } from "../types";
+import { ThemedTitle } from "../title";
+
+const { SubMenu } = Menu;
+const { useToken } = theme;
+
+export const ThemedSider: React.FC<RefineThemedLayoutSiderProps> = ({
```
Comment (omeraplak) [inline packages/antd/src/components/themedLayout/sider/index.tsx:34]: Could we share logic with the Sider component?

File: packages/antd/src/components/themedLayout/header/index.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import { RefineThemedLayoutHeaderProps } from "../types";
+
+const { Text } = Typography;
+const { useToken } = theme;
+
+export const ThemedHeader: React.FC<RefineThemedLayoutHeaderProps> = () => {
+    const { token } = useToken();
+
+    const authProvider = useActiveAuthProvider();
+    const { data: user } = useGetIdentity({
```
Comment (omeraplak) [inline packages/antd/src/components/themedLayout/header/index.tsx:13]: Could we share logic with the Header component?

File: packages/antd/src/components/themedLayout/title/index.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        />
+    </svg>
+);
+
+export const ThemedTitle: React.FC<RefineLayoutThemedTitleProps> = ({
+    collapsed,
+    icon = defaultIcon,
+    text = defaultText,
+}) => {
+    const { token } = useToken();
```
Comment (omeraplak) [inline packages/antd/src/components/themedLayout/title/index.tsx:36]: Could we share logic with the Title component?

File: packages/antd/src/definitions/themes/index.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import { ThemeConfig } from "antd";
+
+export const BLUE_THEME: ThemeConfig = {
```
Comment (omeraplak) [inline packages/antd/src/definitions/themes/index.ts:3]: Why did we choose a distant naming convention for humans? I think we can extract names like Blue, Purple from an object like RefineThemes. This way, we can use them in IntelliSense.


ex:
```ts
import { RefineThemes } from "@refinedev/core";

    const [config, setConfig] = useState({
        token: RefineThemes.Blue.token,
        algorithm: theme.darkAlgorithm,
    });
```

**Review by omeraplak — CHANGES_REQUESTED** [review]

File: examples/theme-antd-demo/package.json [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    "@refinedev/simple-rest": "^4.1.0",
+    "@types/node": "^12.20.11",
+    "@types/react": "^18.0.0",
+    "@types/react-dom": "^18.0.0",
+    "react": "^18.0.0",
+    "react-dom": "^18.0.0",
+    "@uiw/react-md-editor": "^3.19.5",
+    "react-scripts": "^5.0.0",
+    "cross-env": "^7.0.3",
+    "typescript": "^4.7.4",
```
Comment (BatuhanW) [inline examples/theme-antd-demo/package.json:19]: I know we had these earlier, but let's add them to devDependencies for new examples.

File: examples/theme-antd-demo/package.json [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  "private": true,
+  "dependencies": {
+    "@refinedev/antd": "^5.1.0",
+    "@refinedev/core": "^4.1.0",
+    "@refinedev/cli": "^2.1.0",
+    "@refinedev/react-router-v6": "^4.1.0",
+    "@refinedev/simple-rest": "^4.1.0",
+    "@types/node": "^12.20.11",
+    "@types/react": "^18.0.0",
+    "@types/react-dom": "^18.0.0",
```
Comment (BatuhanW) [inline examples/theme-antd-demo/package.json:13]: I know we had these earlier, but let's add them to devDependencies for new examples.

File: packages/antd/src/components/themedLayout/sider/index.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+                )
+            ) {
+                return;
+            }
+        });
+    });
+
+    // layoutSiderTests.bind(this)(Sider);
+
+    xit("should work sider collapse ", async () => {
```
Comment (BatuhanW) [inline packages/antd/src/components/themedLayout/sider/index.spec.tsx:22]:
```suggestion
    it("should work sider collapse ", async () => {
```

File: packages/antd/src/hooks/layout/useSider/index.tsx [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+const { SubMenu } = Menu;
+
+type Props = {
+    meta?: RefineLayoutSiderProps["meta"];
+    Title?: RefineLayoutSiderProps["Title"];
+    render?: RefineLayoutSiderProps["render"];
+};
+
+export const useSider = ({ Title: TitleFromProps, render, meta }: Props) => {
+    const [collapsed, setCollapsed] = useState<boolean>(false);
```
Comment (omeraplak) [inline packages/antd/src/hooks/layout/useSider/index.tsx:36]: Could we add some tests?

File: packages/antd/src/components/pages/auth/components/pageTitle/index.tsx [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    text,
+    wrapperStyle,
+}) => {
+    const { token } = useToken();
+
+    if (icon === null && text === null) {
+        return null;
+    }
+
+    const iconToRender = icon === null ? null : icon || defaultIcon;
```
Comment (BatuhanW) [inline packages/antd/src/components/pages/auth/components/pageTitle/index.tsx:47]:
```suggestion
    const iconToRender = icon ?? defaultIcon;
```

If I'm not missing something here, I think this suggested change is easier to read.

File: packages/antd/src/components/pages/auth/components/pageTitle/pageTitle.spec.tsx [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    });
+
+    test("should render default icon", () => {
+        const { getByTestId } = render(<AuthPageTitle />);
+        expect(getByTestId("auth-page-title-icon")).toBeInTheDocument();
+    });
+
+    test("should render custom icon", () => {
+        const { getByTestId } = render(
+            <AuthPageTitle icon={<div data-testid="custom-icon" />} />,
```
Comment (BatuhanW) [inline packages/antd/src/components/pages/auth/components/pageTitle/pageTitle.spec.tsx:29]: 🔥

**Review by omeraplak — APPROVED** [review]
🥇

**Review by yildirayunlu — APPROVED** [review]

**Review by BatuhanW — APPROVED** [review]

**Pull request merged**